docs/BR.md

```diff
@@ -2051,19 +2073,25 @@ Subject attributes MUST NOT contain only metadata such as '.', '-', and ' ' (i.e
+
+* `dNSName`: The entry MUST contain either a Fully-Qualified Domain Name or Wildcard Domain Name that the CA has validated in accordance with [Section 3.2.2.4](#3224-validation-of-domain-authorization-or-control). Wildcard Domain Names MUST be validated for consistency with [Section 3.2.2.6](#3226-wildcard-domain-validation). The entry MUST NOT contain an Internal Name.
 
-Wildcard FQDNs are permitted.
+   The Fully-Qualified Domain Name or the FQDN portion of the Wildcard Domain Name contained in the entry MUST be composed entirely of LDH Labels joined together by a U+002E FULL STOP (".") character. The zero-length Domain Label representing the root zone of the Internet Domain Name System MUST NOT be included (e.g. "example.com" MUST be encoded as "example.com" and MUST NOT be encoded as "example.com.").
 
-CAs SHALL NOT issue certificates with a `subjectAltName` extension or `subject:commonName` field containing a Reserved IP Address or Internal Name.
+   Effective 2021-10-01, the Fully-Qualified Domain Name or the FQDN portion of the Wildcard Domain Name MUST consist solely of Domain Labels that are P-Labels or Non-Reserved LDH Labels.
 
-Entries in the `dNSName` MUST be in the "preferred name syntax", as specified in RFC 5280, and thus MUST NOT contain underscore characters ("_").
+* `iPAddress`: The entry MUST contain an IPv4 or IPv6 address that the CA has validated in accordance with [Section 3.2.2.5](#3225-authentication-for-an-ip-address). The entry MUST NOT contain a Reserved IP Address.
 
 ##### 7.1.4.2.2 Subject Distinguished Name Fields
 
 a. __Certificate Field:__ `subject:commonName` (OID 2.5.4.3)  
    __Required/Optional:__ __Deprecated__ (Discouraged, but not prohibited)  
-   __Contents:__ If present, this field MUST contain a single IP address or Fully-Qualified Domain Name that is one of the values contained in the Certificate's `subjectAltName` extension (see [Section 7.1.4.2.1](#71421-subject-alternative-name-extension)).
+   __Contents:__ If present, this field MUST contain exactly one entry that is one of the values contained in the Certificate's `subjectAltName` extension (see [Section 7.1.4.2.1](#71421-subject-alternative-name-extension)). The value of the field MUST be encoded as follows:
+
+   * If the value is an IPv4 address, then the value MUST be encoded as an IPv4Address as specified in RFC 3986, Section 3.2.2.
+   * If the value is an IPv6 address, then the value MUST be encoded in the text representation specified in RFC 5952, Section 4.
+   * If the value is a Fully-Qualified Domain Name or Wildcard Domain Name, then the value MUST be encoded as a character-for-character copy of the `dNSName` entry value from the `subjectAltName` extension. Specifically, all Domain Labels of the Fully-Qualified Domain Name or FQDN portion of the Wildcard Domain Name must be encoded as LDH Labels, and P-Labels MUST NOT be converted to their Unicode representation.
```

Comment:
Hi all, can I just ask for a clarification on this line.  Does this mean that the commonName field must be _exactly_ found in a SAN dNSName entry or would it be reasonable for the common name to be EXAMPLE.COM and the SAN dNSName to be example.com?  DNS names are not case-sensitive so it shouldn't matter but I just want to clarify what precisely is meant by a character-for-character copy?

Reply:
Thanks for raising this. I agree this isn't clear and I should have worded this better.
As you mentioned, DNS is case-insensitive so differences in casing aren't material. Given this, I think it's reasonable to interpret this as a "case-insensitive character-for-character copy". Nonetheless, we should make this explicit in the next "Spring cleanup" ballot.

Reply:
Yes, I agree, we should interpret as @CBonnell recommends and fix in a clean up ballot.

Reply:
I'm not sure I agree with Corey here with how we should interpret it. A character in an IA5String is an unambiguous value from the ASCII/IA5 character table, and an 'a' character is not equivalent to an 'A' character.

I don't believe our current language permits this case differential. I understand Corey reasonably suggesting we can consider changing that, but I don't believe that's currently what the text reads.

Reply:
The intent of my writing "character-for-character copy" was to prohibit transformation of P-labels in a SAN dNSName to U-labels in the CN. The next sentence, "Specifically, " clarifies that is the intent. Even if we did mandate a case-sensitive match, I'm not sure what goal that would accomplish from a practical standpoint.

Reply:
The element of "character-for-character" is still relevant, because a CN is not an IA5String, but a DirectoryString.  If you're using the allowances in RFC 4.1.2.6 ( https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.6 ), then while the PrintableString and UTF8String encodings for the characters remain the same as IA5String, different encodings are used if you're using BMPString/UniversalString.

The existing BRs have the language:
> If present, this field MUST contain a single IP address or Fully‐Qualified Domain Name that is one of the values contained in the Certificate’s subjectAltName extension 

And, just like eliminating ambiguity about whether a U-Label is "contained" in a P-label-expressed `dNSName`, I thought the explicit goal here was to ensure a "byte for byte" copy, while accounting for encoding - thus, "character for character". This was certainly the understanding when endorsing, because I'm not sure how we can say "a" is a "character by character copy". It's considered equivalent under DNS matching, yes, but that is not a copy. Allowing 'a' for 'A' is functionally no different than allowing a U-label to be a substitution of the A-label, by arguing that it is a "character by character" copy, simply expressed as a U-label.

There is no legitimate need to express the CN in a separate case syntax than a SAN, and the mere existence of such a permutation is indicative of a point of concern.

Reply:
> Allowing 'a' for 'A' is functionally no different than allowing a U-label to be a substitution of the A-label, by arguing that it is a "character by character" copy, simply expressed as a U-label.

A U-label will by definition not be a LDH label. So the P-label to U-label conversion case is prohibited by this language.

> There is no legitimate need to express the CN in a separate case syntax than a SAN, and the mere existence of such a permutation is indicative of a point of concern.

I think I agree it seems "cleaner" from an engineering purity standpoint that they are binary equal, but I'm not sure of the practical benefit of mandating. Is there a concern about confusable characters or something similar that such case mismatches would cause?

Reply:
> So the P-label to U-label conversion case is prohibited by this language.

Just to make sure we're on the same page: I *believe* we're saying that, even if we drop the "Specifically" language, the requirement still exists that P-label to U-Label conversions are prohibited. I'm saying that with, or without, the "Specifically" language, case conversions are *similarly* prevented, on the same principle: they are character transformations, and thus *not* character-by-character copies.

> Is there a concern about confusable characters or something similar that such case mismatches would cause?

You're going for the "why not" (default allow), I'm going from the "why" (default deny). There is no legitimate need for this flexibility, and the existence of this flexibility can easily lead to implementation bugs by the CA. That is, if the value of the CN and SAN are both populated by the CA by their systems, then there's no reason to ever expect a case mismatch. The only time one should expect a case mismatch is if one, or both, fields come from user-supplied or manually-entered data. While it's _possible_ that the CA performed the necessary case-folding comparison to ensure this requirement is met, it is unfortunately both possible and demonstrably likely that the CA merely transposed one or both values from user-supplied data. Such transposition can easily lead to issues (e.g. if a CA mistakenly uses `str{n}casecmp` to compare the CN and SAN, while failing to account for the possibility of embedded `NUL`s in the ASN.1 string).

This is why I say that this flexibility is intrinsically bad: the only time it's useful is for when a CA is doing something that is known error prone. I don't believe the current language permits that badness, on the same basis that just because certain Unicode characters may compare equivalent does not mean that they are identical characters/codepoints (which is why we require Unicode normalization, by way of RFC 5280).

Reply:
> Just to make sure we're on the same page: I believe we're saying that, even if we drop the "Specifically" language, the requirement still exists that P-label to U-Label conversions are prohibited. I'm saying that with, or without, the "Specifically" language, case conversions are similarly prevented, on the same principle: they are character transformations, and thus not character-by-character copies.

I agree that as presently worded, it can be read that case mismatches are prohibited. However, my intent wasn't to prohibit that and was merely to prohibit conversion to U-labels.

To be clear: are you asserting that the Chrome Root Program views such case mismatches as mis-issuance when the ballot clears IPR?

> There is no legitimate need for this flexibility, and the existence of this flexibility can easily lead to implementation bugs by the CA

I agree that such case mismatches are a code smell, absent someone providing a compelling reason that such mismatches are valuable. That being said, this prohibition would affect many CAs with short notice. If that were the intent, I would have proposed a future effective date to give time to transition, especially since this wasn't a previously stated expectation by Root Programs.

Reply:
> To be clear: are you asserting that the Chrome Root Program views such case mismatches as mis-issuance when the ballot clears IPR?

Yes, we will view this as misissuance, especially since we felt the (new) wording was unambiguous in its expectations, even if that may not have been what you personally intended.

I think the best path forward here would be to discuss with the broader SCWG, particularly to solicit feedback from other root programs for their perspectives.

As I mentioned, while you view this as a new requirement, this seems to be a clarification of the existing requirement. While I can understand your intent may have been different, I'm also not sure how there can be any supported interpretation other than what I mentioned. In the broader conversation, it may be useful to explain how to support the case-insensitive interpretation (not just because of intent), as that will help better inform our handling of any misissuances by CAs.

We've been here before with other ballots, so I don't want to seem inflexible: we handle misissuances and failures to revoke based on the information available to us, so if, for example, a CA fails to adopt the expected action, because of a legitimate misinterpretation of the requirement, it's useful for us to understand how the misinterpretation came to be, so we can both correct it and prevent such issues in the future.

Reply:
Ryan, I assume you would potentially be looking at certs issued after the ballot passes IPR review as mis-issued?  Or would consider all certificates mis-issued regardless of issue date?

Reply:
@mike-reilly Effective date (i.e [~3 weeks from now](https://cabforum.org/2021/07/22/ballot-sc48-v2-domain-name-and-ip-address-encoding/))

Reply:
Understand.  Certs would be considered mis-issued by Google Chrome after effective date (about 3 weeks from now).  Thanks, Mike

Reply:
To clarify......only certs issued **AFTER** the IPR review ends would be potentially be considered mis-issued by Google Chrome correct?

Reply:
Correct. Certificates issued after the effective date are misissued if they are not exact character by character copies (that is, preserving case)